torch/_dynamo/eval_frame.py

```diff
@@ -117,11 +117,11 @@ def innermost_fn(fn):
 
 
 @contextlib.contextmanager
-def enable_dynamic(enable: bool = True):
-    if not enable:
+def set_dynamic(dynamic: bool, specialize_int_float: bool):
+    if not dynamic:
         yield
         return
-    with config.patch(dynamic_shapes=True, specialize_int_float=False):
+    with config.patch(dynamic_shapes=True, specialize_int_float=specialize_int_float):
```

Comment:
Setting dynamic should not set it to False, or True - just remove this? The idea is, I think, that dynamic doesn't touch it. Instead, its up to other regimes (eager vs export) to set the flag.